src/flash.h

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2021 Golioth, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef __APP_FLASH_H__
+#define __APP_FLASH_H__
+
+#ifdef CONFIG_BOOTLOADER_MCUBOOT
+
+#include <zephyr/dfu/flash_img.h>
+#include <zephyr/dfu/mcuboot.h>
+#include <zephyr/types.h>
+
+int flash_img_prepare(struct flash_img_context *flash);
+
+#else /* CONFIG_BOOTLOADER_MCUBOOT */
+
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+
+struct flash_img_context {
+	/* empty */
+};
```

Comment:
```suggestion
struct flash_img_context;
```
Where is the concrete definition for this struct?

Reply:
This fragment is true only for `CONFIG_BOOTLOADER_MCUBOOT=n` case. MCUboot is disabled when **sysbuild** is not used. This is true mostly for testing and with `native_sim` platform, which does not have support for bootloader builds and chain-loading application image.

Note: this file is copy-pasted from https://github.com/golioth/golioth-zephyr-sdk/blob/main/samples/dfu/src/flash.h (which is BTW why 2021 is the year in the copyright statement).